### ISSUE TITLE
recording from two cameras at the same time error

### ISSUE THREAD

**Issue by tareqalbeesh** [pedroSG94/RootEncoder#862]
I am trying to record from both front camera and the back camera at the same time, 
that resulted in producing two videos, 
both recorded the scene right, 
I recorded 12 seconds from both cameras at the same time,
in the result both had the right recorded videos, but the second video always have bigger duration (1 hour 19 minutes in this test and sometimes it gives other random numbers like 23 hours 30 minutes), and this duration is not real but the video player shows it, and the video size is the same as the other video, and in the huge duration video the first 12 seconds are rightly recorded and the rest of the video is corrupted. 
here's two Images that compare both the video recorded from the first camera and the video recorder from the back camera (at the same time for starting the record and stopping the record):
1- the back camera recorded video:
![Background Camera](https://user-images.githubusercontent.com/20373185/118481660-636d3200-b71c-11eb-974e-59516a51088f.png)

2- the front camera recorded video:
![Front Camera](https://user-images.githubusercontent.com/20373185/118481717-741da800-b71c-11eb-9b60-2aa25df6679b.png)

I should also mention that I am in this sample opening the front camera first then the back camera,
if I did the opposite the result would be opposite, meaning if I open the back camera first then the front camera second, the corrupted video would be the video recorded from the front camera,
in short,
record front -> record back -> corrupted video is the back video
record back -> record front -> corrupted video is the front video

**Comment by pedroSG94** [comment]
Check that bufferInfo value is correctly setted. Mostly presentationTimeUs.

**Comment by tareqalbeesh** [comment]
after doing some debugging I got to the reason why is that happening,
when I stop recording the method `stopRecord` would call `stopStream` in case of not streaming and that what is causing the resulted video to be the way it is. 
removing `stopStream  `resulted in perfect two videos, but another problem is showing now: 
`Error clearing streaming request: Function not implemented (-38)` 
but everything stays alright, but I am afraid that it would make a serious problem on the long

**Comment by pedroSG94** [comment]
stopStream is necessary to stop encoders.
I recommend you check that method and find line reason.
The only call of recordController in that method is this:
https://github.com/pedroSG94/rtmp-rtsp-stream-client-java/blob/master/rtplibrary/src/main/java/com/pedro/rtplibrary/base/Camera2Base.java#L644
Maybe a good start could be remove that method to know if it is realted with stop encoders
